lib/Dialect/TritonNvidiaGPU/Transforms/RewriteTensorPointer.cpp

```diff
@@ -615,6 +618,77 @@ class TritonGPURewriteTensorPointerPass
     return nullptr;
   }
 
+  Operation *rewriteIfOp(OpBuilder &builder, scf::IfOp op,
+                         std::stack<Operation *> &eraser,
+                         DenseSet<Value> &valueToRemove) {
+    auto thenYieldOp = op.thenYield();
+    assert(op.getNumResults() == thenYieldOp.getNumOperands());
+    SmallVector<Value> results = thenYieldOp.getOperands();
+
+    // get new result types for IfOp
+    SmallVector<Type> newRetTypes;
+    for (unsigned i = 0; i < results.size(); ++i) {
+      if (!tt::isTensorPointerType(results[i].getType()) ||
+          !valueToRemove.count(results[i])) {
+        newRetTypes.push_back(results[i].getType());
+        continue;
+      }
+      auto makeTensorPtrOp = getMakeTensorPtrOp(results[i]);
+      assert(rewritedInfo.count(makeTensorPtrOp.getResult()));
+      auto info = rewritedInfo[makeTensorPtrOp.getResult()];
+      for (unsigned j = 0; j < info.length(); ++j) {
+        newRetTypes.push_back(builder.getI64Type());
+      }
+    }
+
+    // create and clone new IfOp
+    bool hasElse = !op.getElseRegion().empty();
+    scf::IfOp newOp = builder.create<scf::IfOp>(op.getLoc(), newRetTypes,
+                                                op.getCondition(), hasElse);
+    IRMapping mapping;
+    for (unsigned i = 0; i < op->getNumOperands(); ++i) {
+      mapping.map(op->getOperand(i), newOp->getOperand(i));
+    }
+    auto rematerialize = [&](Block *block) {
+      for (Operation &opInIf : block->getOperations()) {
+        auto newOp = builder.clone(opInIf, mapping);
+        for (unsigned i = 0; i < opInIf.getNumResults(); ++i) {
+          if (valueToRemove.count(opInIf.getResult(i)))
+            valueToRemove.insert(newOp->getResult(i));
```

Comment:
It's not safe if the onInIf has regions. better to re-use the implementation for scf.for op.

Reply:
done